src/main/java/org/jboss/remoting3/ConnectionPeerIdentityContext.java

```diff
@@ -468,7 +468,7 @@ public ConnectionPeerIdentity getCurrentIdentity() {
 
         private final int id;
         private byte[] saslBytes;
-        private int status;
+        private volatile int status;
```

Comment:
@dmlloyd shouldn't this field be volatile? Maybe even a an AtomicFieldUpdater?

Reply:
It's not necessary as all accesses to the field should already be protected by `synchronized` blocks.